tests/jballerina-unit-test/src/test/resources/test-src/lock/lock-on-fail.bal

```diff
@@ -62,7 +62,46 @@ function checkLockWithinLock() returns [int, string] {
     return [lockWithinLockInt1, lockWithinLockString1];
 }
 
+function onFailLockWithinLockWithoutVariable() {
+    lock {
+        lockWithinLockInt1 = 50;
+        lockWithinLockString1 = "sample value";
+        lock {
+            lockWithinLockString1 = "second sample value";
+            lockWithinLockInt1 = 99;
+            lock {
+                lockWithinLockInt1 = 90;
+            }
+            error err = error("custom error", message = "error value");
+            fail err;
+        }
+    } on fail {
+        lockWithinLockInt1 = 100;
+        lockWithinLockString1 = "Error caught";
+    }
+
+    assertEquality(100, lockWithinLockInt1);
+    assertEquality("Error caught", lockWithinLockString1);
+}
+
 function getError()  returns int|error {
     error err = error("Custom Error");
     return err;
-}
\ No newline at end of file
+}
+
+type AssertionError error;
+
+function assertEquality(any|error expected, any|error actual) {
```

Comment:
Atm, we only seem to be calling this with `anydata` values. So I would use `anydata` parameter types and simplify the function.